portal-impl/src/com/liferay/portal/service/impl/CountryServiceImpl.java

```diff
@@ -68,60 +90,237 @@ public Country addCountry(
 
 		Country country = countryPersistence.create(countryId);
 
-		country.setName(name);
 		country.setA2(a2);
 		country.setA3(a3);
-		country.setNumber(number);
-		country.setIdd(idd);
 		country.setActive(active);
+		country.setIdd(idd);
+		country.setName(name);
+		country.setNumber(number);
 
 		return countryPersistence.update(country);
 	}
 
+	@Override
+	public void deleteCountry(long countryId) throws PortalException {
+		if (!getPermissionChecker().isOmniadmin()) {
+			throw new PrincipalException.MustBeOmniadmin(
+				getPermissionChecker());
+		}
+
+		countryLocalService.deleteCountry(countryId);
+	}
+
 	@Override
 	public Country fetchCountry(long countryId) {
 		return countryPersistence.fetchByPrimaryKey(countryId);
 	}
 
+	/**
+	 * @deprecated As of Cavanaugh (7.4.x)
+	 */
+	@Deprecated
 	@Override
 	public Country fetchCountryByA2(String a2) {
 		return countryPersistence.fetchByA2(a2);
 	}
 
+	/**
+	 * @deprecated As of Cavanaugh (7.4.x)
+	 */
+	@Deprecated
 	@Override
 	public Country fetchCountryByA3(String a3) {
 		return countryPersistence.fetchByA3(a3);
 	}
 
+	@Override
+	public Country fetchCountryByCompanyIdAndA2(long companyId, String a2) {
+		return countryPersistence.fetchByC_A2(companyId, a2);
+	}
+
+	@Override
+	public Country fetchCountryByCompanyIdAndA3(long companyId, String a3) {
+		return countryPersistence.fetchByC_A3(companyId, a3);
+	}
+
+	@Override
+	public int getCountByCompanyIdAndActive(long companyId, boolean active)
+		throws PortalException {
+
+		if (!getPermissionChecker().isOmniadmin()) {
+			throw new PrincipalException.MustBeOmniadmin(
+				getPermissionChecker());
+		}
+
+		return countryLocalService.getCountriesCountByCompanyIdAndActive(
+			companyId, active);
+	}
+
 	@Override
 	public List<Country> getCountries() {
 		return countryPersistence.findAll();
 	}
 
+	/**
+	 * @deprecated As of Cavanaugh (7.4.x)
+	 */
 	@AccessControlled(guestAccessEnabled = true)
+	@Deprecated
 	@Override
 	public List<Country> getCountries(boolean active) {
 		return countryPersistence.findByActive(active);
 	}
 
+	@Override
+	public List<Country> getCountriesByCompanyId(long companyId) {
+		return countryPersistence.findByCompanyId(companyId);
+	}
+
+	@Override
+	public List<Country> getCountriesByCompanyId(
+			long companyId, int start, int end,
+			OrderByComparator<Country> orderByComparator)
+		throws PortalException {
+
+		if (!getPermissionChecker().isOmniadmin()) {
+			throw new PrincipalException.MustBeOmniadmin(
+				getPermissionChecker());
+		}
```

Comment:
It doesn't really make sense that `getCountriesByCompanyId(long companyId)` is not permission checked, but this method is. I think we can just not permission check `get` and `fetch` methods for now. Only permission check the `write` methods.